comprl/server/networking.py

```diff
@@ -0,0 +1,256 @@
+        """
+
+        def callback(res):
+            if res["version"] == VERSION:
+                return_callback(res["token"].decode())
+            else:
+                log.error("Client with wrong version tried to authenticate.")
+                self.send_error(msg="Tried to connect with wrong version")
+                self.transport.loseConnection()
+
+        self.callRemote(Auth).addCallback(callback=callback).addTimeout(
+            ConfigProvider.get("timeout"), reactor, self.connectionTimeout
+        )
+
+    def is_ready(self, return_callback: Callable[[bool], None]) -> bool:
+        """checks if the player is ready to play
+
+        Returns:
+            bool: returns true if the player is ready to play
+        """
+        return (
+            self.callRemote(Ready)
+            .addCallback(callback=lambda res: return_callback(res["ready"]))
+            .addTimeout(ConfigProvider.get("timeout"), reactor, self.connectionTimeout)
+        )
+
+    def notify_start(self, game_id: GameID) -> None:
+        """starts the game
+
+        Args:
+            game (Game): game that starts
+        """
+        return self.callRemote(StartGame, game_id=game_id.bytes)
+
+    def get_step(
+        self, obv: list[float], return_callback: Callable[[list], None]
+    ) -> None:
+        """performs step requested by player"""
+
+        return (
+            self.callRemote(Step, obv=obv)
+            .addCallback(callback=lambda res: return_callback(res["action"]))
+            .addTimeout(ConfigProvider.get("timeout"), reactor, self.connectionTimeout)
+        )
+
+    def notify_end(self, result, stats) -> None:
+        """ends the game"""
+
+        return self.callRemote(EndGame, result=result, stats=stats).addTimeout(
+            ConfigProvider.get("timeout"), reactor, self.connectionTimeout
+        )
+
+    def send_error(self, msg: str):
+        """send an error string to the client"""
+        self.callRemote(Error, msg=str.encode(msg))
+
+    def disconnect(self):
+        """disconnects the client from the server"""
+        self.transport.loseConnection()
+
+
+class COMPPlayer(IPlayer):
+    """player of the game"""
+
+    def __init__(self, connection: COMPServerProtocol) -> None:
+        # init super to obtain id
+        super().__init__()
+
+        # set the networking connection
+        self.connection: COMPServerProtocol = connection
+
+    def authenticate(self, result_callback):
+        """authenticates player
+
+        Args: result_callback (callback function)
+
+        Returns: token (string)"""
+        self.connection.get_token(result_callback)
+
+    def is_ready(self, result_callback) -> bool:
+        """checks if the player is ready to play
+
+        Returns:
+            bool: returns true if the player is ready to play
+        """
+        return self.connection.is_ready(result_callback)
+
+    def notify_start(self, game_id: GameID):
+        """notifies start of game"""
```

Comment:
There are defenitly some inconsistencies with the docstrings in the networking because I changed the signatures to many times, will double check. Although I can't see the problem with the notify start

Reply:
It is nothing big, the argument is not described in the docstring. It just stood out to me because all the other methods in this class do this.